source/components/ch4_component.cpp

```diff
@@ -104,25 +104,22 @@ void CH4Component::setData( const string& varName,
     try {
          if( varName ==  D_PREINDUSTRIAL_CH4  ) {
             H_ASSERT( data.date == Core::undefinedIndex() , "date not allowed" );
-            M0 = unitval::parse_unitval( data.value_str, data.units_str, U_PPBV_CH4 );
```

Comment:
So, as I understand it, you are trying to create an alternative front end for Hector, and you want to be able to initialize all of the Hector components _without_ having them read the ini files.  I ran into a similar problem when coupling Hector to GCAM, and in the interest of getting something working quickly, I just punted and had Hector parse an ini file I had created specifically for use with GCAM, but that's not a great solution.

I think the right thing to do here (i.e., what I should have done when I first did the GCAM coupling) is to add some initialization functions to the API.  We could allow these to accept arrays of doubles, with the unit specified just once for the whole array.  We would probably also want to make some provision for designating one (or more?) of the arrays as a key that indexes the others (e.g., the year column in our emissions tables).  We might also arrange for these parts of the API to be disabled once the simulation is running, which would prevent someone from trying to use it inappropriately.

Thoughts?